### ISSUE TITLE
SQL clients not binding to named virtual server

### ISSUE THREAD

**Issue by matracine** [FreeRADIUS/freeradius-server#3770]
# Defect

# Defect/Feature description
Running Freeradius 3.0.21-1 in Debian Buster from source list : 
     deb https://packages.networkradius.com/releases/debian-buster buster main
Configured with mysql connector (freeradius-mysql 3.0.21-1 installed)
If the server is anonyous (no name behind the server {... configuration directive)  and set NULL in server entry of the nas table, the clients are correctly recognised.
If the server is named, say srv1 (server srv1 {... configuration directive) and set "srv1" in server entry of the nas table, the clients are not recognised, got error :
```
freeradius[5927]: Ignoring request to auth address * port 1812 bound to server srv1 from unknown client x.x.x.x port 58368 proto udp
```
The configuration is working correctly when nas definition is done in clients.conf.

I found similars issues googling, but no real solution exept "the default sserver is anonymous so put NULL in server column of the database".
In my case, server is not anonymous, I have 2 servers configured : 
srv1 : udp/port 1812 and udp/ort 1813
srv2 : udp/port 1822 and udp/ort 1823

## How to reproduce issue

Empty clients.conf file.

Added name srv1 to virtual-site section and removed clients directive :
```
   server srv1 {
...
#     clients = clients_pool
...
   }
```

Added clients in nas table :
```
+----+----------------+-----------+-------+-------+---------------------+--------+-----------+---------------+
| id | nasname        | shortname | type  | ports | secret              | server | community | description   |
+----+----------------+-----------+-------+-------+---------------------+--------+-----------+---------------+
|  1 | 1.2.3.4        | NAS1      | other |  NULL | MySecret1           | srv1   | NULL      | RADIUS Client |
|  2 | 1.2.3.5        | NAS2      | other |  NULL | MySecret2           | srv1   | NULL      | RADIUS Client |
+----+----------------+-----------+-------+-------+---------------------+--------+-----------+---------------+
```
Note : IP are faked

In mods-enabled/sql added :
```
read_clients = yes
```

## Output of ``[radiusd|freeradius] -X`` showing issue occurring

```text
...
freeradius[5927]: rlm_sql (sql): Processing generate_sql_clients
freeradius[5927]: rlm_sql (sql) in generate_sql_clients: query is SELECT id, nasname, shortname, type, secret, server FROM nas
freeradius[5927]: rlm_sql (sql): Reserved connection (0)
freeradius[5927]: rlm_sql (sql): Executing select query: SELECT id, nasname, shortname, type, secret, server FROM nas
freeradius[5927]: rlm_sql (sql): Adding client 1.2.3.4 (NAS1) to srv1 clients list
freeradius[5927]: rlm_sql (1.2.3.4): Client "NAS1" (sql) added
freeradius[5927]: rlm_sql (sql): Adding client 1.2.3.5 (NAS2) to srv1 clients list
freeradius[5927]: rlm_sql (1.2.3.5): Client "NAS2" (sql) added
freeradius[5927]: rlm_sql (sql): Released connection (0)
...

freeradius[5927]: Listening on auth address * port 1822 bound to server srv2
freeradius[5927]: Listening on acct address * port 1823 bound to server srv2
freeradius[5927]: Listening on auth address * port 1812 bound to server srv1
freeradius[5927]: Listening on acct address * port 1813 bound to server srv1
..
freeradius[5924]: Ignoring request to auth address * port 1812 bound to server srv1 from unknown client 1.2.3.4 port 39923 proto udp
```

**Comment by alandekok** [comment]
Please check the v3.0.x branch.  I believe that commit a4332241 may fix this already.

**Comment by matracine** [comment]
Hello Alan,
Thanks,but I'm running latests 3.0.x version (I upgraded to 3.0.21-1 when I detected the problem with the standard Debian version).
Is this commit merged in this release ? (Not a pro with github to see that)

On the other hand,  the code seems related to detecting an error while inserting a new client.
I do not have this kind of error and my clients seems inserted correctly : 
```
freeradius[5927]: rlm_sql (sql): Adding client 1.2.3.4 (NAS1) to srv1 clients list
freeradius[5927]: rlm_sql (sql): Adding client 1.2.3.5 (NAS2) to srv1 clients list
```
But when a request appears, they are not found.
```
freeradius[5924]: Ignoring request to auth address * port 1812 bound to server srv1 from unknown client 1.2.3.4
```

Regards,

**Comment by alandekok** [comment]
You may be running the latest *released* version, but changes have been made since them.  If you look at the commit I mentioned, it was done in July.  *After* 3.0.21 was released.  Therefore, the fix isn't in 3.0.21.

Please try downloading and building the v3.0.x branch from github.  It is the *only* way to verify that the fix has been made, and is correct.

**Comment by matracine** [comment]
Sorry, do not have this possibility.
I will wait for the 3.0.22 to be released and test it...
Thanks for your time,
Regards

**Comment by alandekok** [comment]
If you don't have time to try this on a test machine, then I don't see any reason to keep this open.

FreeRADIUS depends on the community for help.  If your role is to report bugs, and then sit back waiting for us to fix & test the bugs, then we also don't have time to help you.

**Comment by alandekok** [comment]
From email:

```
4- with an empty client.conf, server named srv1 and srv1 in server field of nas table, I still have the error, despite the configuration seems to load correctly :
Nov 27 09:04:08 builder21 freeradius[28026]: rlm_sql (sql): Executing select query: SELECT id, nasname, shortname, type, secret, server FROM nas
Nov 27 09:04:08 builder21 freeradius[28026]: rlm_sql (sql): Adding client 127.0.0.1 (local) to srv1 clients list
Nov 27 09:04:08 builder21 freeradius[28026]: rlm_sql (127.0.0.1): Client "local" (sql) added
...

Nov 27 09:04:34 builder21 freeradius[28026]: Ignoring request to auth address * port 1812 bound to server srv1 from unknown client 127.0.0.1 port 58727 proto udp
```

it's not at all clear what else is going on.  Is the `default` virtual server gone?  renamed?  What is the FULL output of `radiusd -X` ???

**Comment by alandekok** [comment]
This is now fixed as of HEAD in v3.0.x.
